apps/src/code-studio/pd/foorm/Foorm.jsx

```diff
@@ -58,26 +69,84 @@ export default class Foorm extends React.Component {
     if (this.props.submitParams) {
       requestData = {...this.props.submitParams, ...requestData};
     }
+    this.setState({
+      statusMessage: 'Your responses are being submitted...',
+      hasError: false,
+      submitting: true
+    });
+    const startTime = Date.now();
     $.ajax({
       url: this.props.submitApi,
       type: 'post',
       dataType: 'json',
       data: requestData
-    });
+    })
+      .done(() => {
+        this.setCompletedStatusMessage(
+          startTime,
+          'Thank you for completing the survey! Your responses saved successfully.',
+          false,
+          null
+        );
+      })
+      .fail(jqXHR => {
+        if (jqXHR.status === 409) {
+          this.setCompletedStatusMessage(
+            startTime,
+            'You already submitted a response. Thank you!',
+            false,
+            null
+          );
+        } else {
+          this.setCompletedStatusMessage(
+            startTime,
+            'An error occurred and we could not save your response.',
+            true,
+            survey
+          );
+        }
+      });
   };
 
+  setCompletedStatusMessage(startTime, statusMessage, hasError, survey) {
+    const elapsedTime = Date.now() - startTime;
+    // if the api request took less than 1 second, wait until 1 second has
+    // passed so spinner does not flash on the screen.
+    const timeout =
+      elapsedTime > SPINNER_WAIT_MS ? 0 : SPINNER_WAIT_MS - elapsedTime;
+    setTimeout(() => {
+      this.setState({
+        statusMessage: statusMessage,
+        hasError: hasError,
+        survey: survey,
+        submitting: false
+      });
+    }, timeout);
+  }
+
   render() {
-    if (this.state.submitted) {
-      return <div>Thank you for submitting</div>;
-    } else {
-      return (
+    return (
+      <div>
         <Survey.Survey
           model={this.surveyModel}
           onComplete={this.onComplete}
           data={this.props.surveyData}
           css={this.customCss}
+          requiredText={'(Required)'}
```

Comment:
When viewing the survey, there is no space before the `(`.  Can we add one here?

Reply:
Never mind, this looked fine after I did a re-seed.